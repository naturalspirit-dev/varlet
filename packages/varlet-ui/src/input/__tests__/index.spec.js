test('test input', async () => {
  console.log('hello')
})
