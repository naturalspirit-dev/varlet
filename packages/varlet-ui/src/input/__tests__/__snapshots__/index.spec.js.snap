// Vitest Snapshot v1

exports[`test input clear 1`] = `
"<div class=\\"var-input var--box\\">
  <div class=\\"var-input__controller\\">
    <div class=\\"var-input__icon\\"></div>
    <div class=\\"var-input__wrap\\">
      <!--v-if--><input class=\\"var-input__input\\" autocomplete=\\"new-password\\" id=\\"var-input-17\\" type=\\"text\\"><label class=\\"var--ellipsis var-input__placeholder var-input--placeholder-hint\\" for=\\"var-input-17\\"></label>
    </div>
    <div class=\\"var-input__icon\\"><i class=\\"var-icon var-icon--set var-icon-close-circle var-input__clear-icon\\" style=\\"transition: transform 0ms; font-size: 14px;\\" var-input-cover=\\"\\"></i></div>
  </div>
  <div class=\\"var-input__line\\">
    <div class=\\"var-input__dot\\"></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`test input focus & blur 1`] = `
"<div class=\\"var-input var--box\\">
  <div class=\\"var-input__controller var-input--focus\\">
    <div class=\\"var-input__icon\\"></div>
    <div class=\\"var-input__wrap\\">
      <!--v-if--><input class=\\"var-input__input\\" autocomplete=\\"new-password\\" id=\\"var-input-1\\" type=\\"text\\"><label class=\\"var--ellipsis var-input--focus var-input__placeholder var-input--placeholder-hint\\" for=\\"var-input-1\\"></label>
    </div>
    <div class=\\"var-input__icon\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"var-input__line\\">
    <div class=\\"var-input__dot var-input--spread\\"></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`test input focus & blur 2`] = `
"<div class=\\"var-input var--box\\">
  <div class=\\"var-input__controller\\">
    <div class=\\"var-input__icon\\"></div>
    <div class=\\"var-input__wrap\\">
      <!--v-if--><input class=\\"var-input__input\\" autocomplete=\\"new-password\\" id=\\"var-input-1\\" type=\\"text\\"><label class=\\"var--ellipsis var-input__placeholder\\" for=\\"var-input-1\\"></label>
    </div>
    <div class=\\"var-input__icon\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"var-input__line\\">
    <div class=\\"var-input__dot\\"></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`test input hint to be false 1`] = `
"<div class=\\"var-input var--box\\">
  <div class=\\"var-input__controller\\">
    <div class=\\"var-input__icon var-input--non-hint\\"></div>
    <div class=\\"var-input__wrap var-input--non-hint\\">
      <!--v-if--><input class=\\"var-input__input\\" autocomplete=\\"new-password\\" id=\\"var-input-13\\" type=\\"text\\"><label class=\\"var--ellipsis var-input__placeholder var-input--placeholder-hidden var-input--placeholder-non-hint\\" for=\\"var-input-13\\"></label>
    </div>
    <div class=\\"var-input__icon var-input--non-hint\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"var-input__line\\">
    <div class=\\"var-input__dot\\"></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`test input maxlength 1`] = `
"<div class=\\"var-input var--box\\">
  <div class=\\"var-input__controller\\">
    <div class=\\"var-input__icon\\"></div>
    <div class=\\"var-input__wrap\\">
      <!--v-if--><input class=\\"var-input__input\\" autocomplete=\\"new-password\\" id=\\"var-input-9\\" type=\\"text\\" maxlength=\\"100\\"><label class=\\"var--ellipsis var-input__placeholder var-input--placeholder-hint\\" for=\\"var-input-9\\"></label>
    </div>
    <div class=\\"var-input__icon\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"var-input__line\\">
    <div class=\\"var-input__dot\\"></div>
  </div>
  <transition-stub>
    <div class=\\"var-form-details\\">
      <div class=\\"var-form-details__message\\"></div>
      <div class=\\"var-form-details__length\\">4/100</div>
    </div>
  </transition-stub>
</div>"
`;

exports[`test input validation 1`] = `
"<div class=\\"var-input var--box\\">
  <div class=\\"var-input__controller var-input--error\\">
    <div class=\\"var-input__icon\\"></div>
    <div class=\\"var-input__wrap\\">
      <!--v-if--><input class=\\"var-input__input var-input--caret-error\\" autocomplete=\\"new-password\\" id=\\"var-input-32\\" type=\\"text\\"><label class=\\"var--ellipsis var-input--error var-input__placeholder var-input--placeholder-hint\\" for=\\"var-input-32\\"></label>
    </div>
    <div class=\\"var-input__icon\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"var-input__line var-input--line-error\\">
    <div class=\\"var-input__dot var-input--line-error\\"></div>
  </div>
  <transition-stub>
    <div class=\\"var-form-details\\">
      <div class=\\"var-form-details__message\\">长度必须大于3</div>
      <div class=\\"var-form-details__length\\"></div>
    </div>
  </transition-stub>
</div>"
`;

exports[`test input validation 2`] = `
"<div class=\\"var-input var--box\\">
  <div class=\\"var-input__controller\\">
    <div class=\\"var-input__icon\\"></div>
    <div class=\\"var-input__wrap\\">
      <!--v-if--><input class=\\"var-input__input\\" autocomplete=\\"new-password\\" id=\\"var-input-32\\" type=\\"text\\"><label class=\\"var--ellipsis var-input__placeholder\\" for=\\"var-input-32\\"></label>
    </div>
    <div class=\\"var-input__icon\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"var-input__line\\">
    <div class=\\"var-input__dot\\"></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`test input validation 3`] = `
"<div class=\\"var-input var--box\\">
  <div class=\\"var-input__controller\\">
    <div class=\\"var-input__icon\\"></div>
    <div class=\\"var-input__wrap\\">
      <!--v-if--><input class=\\"var-input__input\\" autocomplete=\\"new-password\\" id=\\"var-input-32\\" type=\\"text\\"><label class=\\"var--ellipsis var-input__placeholder var-input--placeholder-hint\\" for=\\"var-input-32\\"></label>
    </div>
    <div class=\\"var-input__icon\\">
      <!--v-if-->
    </div>
  </div>
  <div class=\\"var-input__line\\">
    <div class=\\"var-input__dot\\"></div>
  </div>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>"
`;
